### ISSUE TITLE
Bug: wrong typescript version displayed in the playground

### ISSUE THREAD

**Issue by DetachHead** [typescript-eslint/typescript-eslint#5244]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Relevant Package

website

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.2&sourceType=module&code=PTAuGcAIBYDoHYBclQE8AOBTSA5A9gHY4CuANqQIYBGpmAPACoB8kAvJA5JgB6iYEATKATKlIAH0jFBmAGYBLApgGQA-JCUA3TACdIyBgFgAUCAgxYADmRosuQiXLVajFu04AySAG8AviZNbbAAxPDw2eyJRZ3pwUB1FAHMJDVEmIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
type Foo = NonNullable<string | null>
```

### Expected Result

typescript version in the sidebar is 4.7.2

### Actual Result

typescript version in the sidebar is displayed as 4.8.0-beta when the actual version is 4.7.2

![image](https://user-images.githubusercontent.com/57028336/175763898-0602f734-a138-462e-9e95-84ec71cf324a.png)

### Versions

n/a this is an issue with the website

**Comment by armano2** [comment]
this issue was on todo board, but I forgot to create ticket

> playground should try to match typescript version while ignoring patches, eg, 4.6.2 should be resolved to 4.6.4 instead of latest (4.7.x)
